libp2p/NodeTable.cpp

```diff
@@ -381,7 +374,7 @@ void NodeTable::dropNode(shared_ptr<NodeEntry> _n)
     {
         Guard l(x_state);
         NodeBucket& s = bucket_UNSAFE(_n.get());
-        s.nodes.remove_if([&_n](weak_ptr<NodeEntry> n) { return n.lock() == _n; });
+        s.nodes.remove_if([&_n](weak_ptr<NodeEntry> _bucketEntry) { return _bucketEntry == _n; });
```

Comment:
Did you mean here that I should pass `weak_ptr` by reference to lambda?

Reply:
Ah you probably meant capturing `_n` by reference doesn't ensure it will live as long as lambda.
But this lambda lives only in this statement, I think it's fine to capture locals by reference.